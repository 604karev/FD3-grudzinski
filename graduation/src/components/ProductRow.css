.element {
    text-align: left;
    display: flex;

}
.element button{
    flex: 0 0 0;
    align-self: center;
}
.element .elementName{
    flex: 1 1 0;
    display: block;
    font-family: Roboto, sans-serif;
    font-size: 18px;
    line-height: 1.5;
    letter-spacing: normal;
    text-align: left;
    color: #0063ec;
    cursor: pointer;

}

