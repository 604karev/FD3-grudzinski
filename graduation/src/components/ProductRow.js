import React, {PureComponent} from 'react';
import './ProductRow.css'
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';


class ProductRow extends PureComponent {

    static propTypes = {
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        select: PropTypes.func.isRequired,
        delete: PropTypes.func.isRequired,
        edit: PropTypes.func.isRequired,
    };


    render() {
        console.log("ProductRow render");
        return (

            <div className='element' onClick={(e) => this.props.select(this.props.id)}>
                <span className='elementName'>{this.props.name}</span>
                <span className='elementEdit' onClick={(e) => this.props.edit(this.props.id, e)}>
                    <FontAwesome
                        name="pencil"
                        style={{fontSize: '24px', color: '#0063ec', padding: '0 10px', cursor:'pointer'}}/></span>

                <span className='elementRemove' onClick={(e) => this.props.delete(this.props.id, e)}>
                    <FontAwesome
                        name="trash"
                        style={{fontSize: '24px', color: '#0063ec', padding: '0 10px', cursor:'pointer'}}
                    /></span>

            </div>
        )
    }
}

export default ProductRow

