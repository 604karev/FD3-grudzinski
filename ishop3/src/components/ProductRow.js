import React, {Component} from 'react';
import './ProductRow.css'
import PropTypes from 'prop-types';


class ProductRow extends Component {

    static propTypes = {
        item: PropTypes.shape({
                id: PropTypes.number.isRequired,
                name: PropTypes.string.isRequired,
                price: PropTypes.string.isRequired,
                img: PropTypes.string.isRequired,
                quantity: PropTypes.number.isRequired,
            }
        ),
        selectedState: PropTypes.number,
        select: PropTypes.func.isRequired,
        delete: PropTypes.func.isRequired,
        edit: PropTypes.func.isRequired,

    };


    render() {
        return (
            <div className='element' onClick={() => this.props.select(this.props.item.id)}>
                <span>{this.props.item.name}</span>
                <button onClick={() => this.props.delete(this.props.item.id)}>Удалить</button>
                <button onClick={() => this.props.edit(this.props.item.id)}>Редактировать</button>
            </div>


        )


    }
}

export default ProductRow

