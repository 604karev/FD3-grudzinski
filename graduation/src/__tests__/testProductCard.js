import React from 'react';
import ProductRow from '../components/ProductRow';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<ProductRow id={''} name="" delete={''} edit={''} select={''}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});