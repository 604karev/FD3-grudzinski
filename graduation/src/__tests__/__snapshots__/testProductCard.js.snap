// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="element"
>
  <span
    className="elementName"
  >
    
  </span>
  <span
    className="elementEdit"
  >
    <span
      aria-hidden={true}
      className="fa fa-pencil"
      style={
        Object {
          "color": "#0063ec",
          "cursor": "pointer",
          "fontSize": "24px",
          "padding": "0 10px",
        }
      }
    />
  </span>
  <span
    className="elementRemove"
  >
    <span
      aria-hidden={true}
      className="fa fa-trash"
      style={
        Object {
          "color": "#0063ec",
          "cursor": "pointer",
          "fontSize": "24px",
          "padding": "0 10px",
        }
      }
    />
  </span>
</div>
`;
